src/rational/q/cmp.rs

```diff
@@ -47,6 +48,33 @@ impl PartialEq for Q {
 // We do not allow division by zero, therefore, this is the case.
 impl Eq for Q {}
 
+impl PartialOrd for Q {
+    /// Compares two [`Q`] values. Used by the `<`, `<=`, `>`, and `>=` operators.
+    ///
+    /// Parameters:
+    /// - `other`: the other value that is used to compare the elements
+    ///
+    /// Returns the [`Ordering`] of the elements.
+    ///
+    /// # Example
+    /// ```
+    /// # use qfall_math::error::MathError;
+    /// use qfall_math::rational::Q;
+    ///
+    /// let a: Q = Q::try_from((&1,&10))?;
+    /// let b: Q = Q::try_from((&2,&10))?;
+    ///
+    /// assert!(a < b);
+    /// assert!(a <= b);
+    /// assert!(b > a);
+    /// assert!(b >= a);
+    /// # Ok::<(), MathError>(())
+    /// ```
```

Comment:
is partialOrd possible for borrowed values? if yes, plead add towards doc and tests